users/tests/test_unit_validators.py

```diff
@@ -44,3 +47,24 @@ def _assert_date_of_birth_is_valid(self, date_of_birth):
             self.user_age_validator(date_of_birth)
         except Exception as e:  # pylint:disable=broad-except
             self.fail(f"Unexpected exception {str(e)} has occurred!")
+
+
+class TestUserEmailValidation(TestCase):
+    def setUp(self) -> None:
+        self.user_email_validation = UserEmailValidation()
+
+    @parameterized.expand(
+        [
+            ("testusercodepoets.it", ValidationErrorText.VALIDATION_ERROR_EMAIL_AT_SIGN_MESSAGE),
+            ("@codepoets.it", ValidationErrorText.VALIDATION_ERROR_EMAIL_MALFORMED_FIRST_PART),
+            ("testuser@invaliddomain.com", ValidationErrorText.VALIDATION_ERROR_EMAIL_MESSAGE_DOMAIN),
+        ]
+    )
+    def test_validator_should_throw_exception_with_invalid_email(self, email, exception_message):
+        with self.assertRaises(ValidationError) as exception:
+            self.user_email_validation(email)
+        self.assertEqual(exception_message, exception.exception.message)
+
+    def test_validator_should_accept_valid_email(self):
+        valid_email = self.user_email_validation("valid_email@codepoets.it")
+        self.assertEqual(valid_email, "valid_email@codepoets.it")
```

Comment:
This is good time for you to learn how `override_settings` decorator works. Please add positive test for different value in `VALID_EMAIL_DOMAIN_LIST` for example that, when this setting will have more than one element in list etc.

Reply:
Done